sysfs-led-main.c

```diff
@@ -623,6 +640,62 @@ sysfs_led_generate_ramp_hard_step(int ms_on, int ms_off)
           sysfs_led_breathe.delay, steps_on, steps_off);
 }
 
+/** Generate triangle wave for use from breathing timer
+ */
+static void
+sysfs_led_generate_ramp_triangle(int ms_on, int ms_off)
+{
+  /* Round up to multiples of minimum step size */
+  int ms_step = SYSFS_LED_STEP_DELAY;
+
+  ms_on  = led_util_roundup(ms_on,  ms_step);
+  ms_off = led_util_roundup(ms_off, ms_step);
+
+  int ms_tot = ms_on + ms_off;
+
+  /* Calculate number of steps */
+  int steps_tot = (ms_tot + ms_step - 1) / ms_step;
+
+  if( steps_tot > SYSFS_LED_MAX_STEPS ) {
+    steps_tot = SYSFS_LED_MAX_STEPS;
+    ms_step = (ms_tot + steps_tot - 1) / steps_tot;
+
+    if( ms_step < SYSFS_LED_STEP_DELAY )
+      ms_step = SYSFS_LED_STEP_DELAY;
+  }
+
+  /* Divide time between raising and lowering edges.
+   * Put rounding excess to raising edge to facilitate
+   * zero ms_off time used for sawtooth case.
+   */
+  int steps_off = (ms_off + ms_step - 1 ) / ms_step;
+  int steps_on  = steps_tot - steps_off;
+
+  /* Set a non-zero intensity value for each step on the ramp.
+   */
+  int i = 0;
+
+  for( ; i < steps_on; ++i )
+    sysfs_led_breathe.value[i] = led_util_trans(i, 0, steps_on, 1, 255);;
+
+  for( ; i < steps_tot; ++i )
+      sysfs_led_breathe.value[i] = led_util_trans(steps_tot - i, 0, steps_off, 1, 255);
+
+  sysfs_led_breathe.delay = ms_step;
+  sysfs_led_breathe.steps = steps_tot;
+
+  mce_log(LL_DEBUG, "delay=%d, steps_on=%d, steps_off=%d",
+          sysfs_led_breathe.delay, steps_on, steps_off);
+}
+
+/** Generate sawtooth wave for use from breathing timer
+ */
+static void
+sysfs_led_generate_ramp_sawtooth(int ms_on, int ms_off)
+{
+    sysfs_led_generate_ramp_triangle(ms_on + ms_off, 0);
```

Comment:
Indentation in rest of the file is 2 spaces not 4

Reply:
fixed